java-vtl-script/src/main/java/no/ssb/vtl/script/operations/AggregationOperation.java

```diff
@@ -76,35 +76,52 @@ protected DataStructure computeDataStructure() {
     @Override
     public Stream<DataPoint> getData() {
         DataStructure childStructure = getChild().getDataStructure();
+
+        Order.Builder orderBuilder = Order.create(childStructure);
+        groupBy.forEach(component -> orderBuilder.put(component, Order.Direction.ASC));
+        Order orderForSorting = orderBuilder.build();
+
+        Stream<DataPoint> data = getChild().getData(orderForSorting)
+                .orElseGet(() -> getChild().getData().sorted(orderForSorting));
+
+        return this.processStream(data);
+    }
+
+    @Override
+    public Optional<Stream<DataPoint>> getData(Order orders, Filtering filtering, Set<String> components) {
```

Comment:
In this method, orders may contain only Country. If we pass orders as is we might get occupation in random order.

Reply:
It is very important to respect the ordering that is passed to this method, if not other operations will fail. In the case of aggregation, the order has to be passed along to the processing function since it relies on the particular order that was requested.